packages/plugin-electron-net-breadcrumbs/net-breadcrumbs.js

```diff
@@ -32,28 +32,36 @@ module.exports = net => ({
         return request
       }
 
+      const originalEnd = request.end
+      let requestStart
+
+      request.end = (...args) => {
+        requestStart = new Date()
+        originalEnd.apply(request, args)
+      }
```

Comment:
we save the `requestStart` date on `request.end`? if that's correct I think a comment here would be nice

Reply:
Aah this is a good point actually - if the request is using chunked encoding that would only be the duration of the final chunk that was sent. Maybe it's better to set `requestStart` on the first write operation instead, or do something like this (cc @imjoehaines):
```suggestion
      let requestStart

      // For chunked requests the request begins when the first chunk is written,
      // otherwise the request begins when the last chunk is written
      const originalWrite = request.write
      request.write = (...args) => {
        if (request.chunkedEncoding && !requestStart) requestStart = new Date()          
        originalWrite.apply(request, args)
      }

      const originalEnd = request.end
      request.end = (...args) => {
        if (!requestStart) requestStart = new Date()
        originalEnd.apply(request, args)
      }
```

Reply:
`request.end` is what finalises the request, i.e. it makes sure that everything is actually sent. This is still a good point though — `request.write` _can_ start the process of sending the request:

> The first write operation ***may*** cause the request headers to be issued on the wire
> https://www.electronjs.org/docs/latest/api/client-request#requestwritechunk-encoding-callback

So only using `request.end` means the duration may be wrong if we want to capture the duration from the point the request actually starts being sent — I doubt it will make a different most of the time in a real app TBH, especially with `Date` only having millisecond precision, but if there's a large request body it could take a non-trivial amount of time from writing it to the request with `request.write` and finishing the request with `request.end`

Maybe we need to use both `request.write` & `request.end`, recording the start time from whichever is called first? You have to call `request.end` but don't necessarily have to call `request.write`, so we can't just use `request.write`

Reply:
Replied at the same time as you @yousif-bugsnag!

What you suggested LGTM if it's only when using chunked encoding that `write` can start the request. Electron's docs aren't that specific so I'm not sure if there are other cases where `write` can start sending the request?

Reply:
Cool, yep, looks that way - I've been doing some manual testing with a local http server and that tallies up with what I've observed (the connection is established on `write` only when `request.chunkedEncoding = true`)

Although looks like I could have just read the source! 😅